public/keycloak/keycloak.ts

```diff
@@ -140,7 +138,7 @@ class KeyCloakService {
       return
     }
     // if isForceRefresh is true, send -1 in updateToken to force update the token
-    const tokenExpiresIn = (isForceRefresh) ? -1 : this.kc.tokenParsed.exp - Math.ceil(new Date().getTime() / 1000) + this.kc.timeSkew + 100
+    const tokenExpiresIn = isForceRefresh  ? -1 : (this.kc?.tokenParsed?.exp || 0) - Math.ceil(new Date().getTime() / 1000) + (this.kc?.tokenParsed?.timeSkew || 0) + 100;
```

Comment:
This is fairly complicated. I recommend simplifying this with a normal if-then-else statement and variables to hold intermediate values.

Reply:
Thanks! It still looks complicated but I think now I can figure it out 😆